common/djangoapps/student/views/dashboard.py

```diff
@@ -624,11 +629,17 @@ def student_dashboard(request):
         )
 
     enterprise_message = get_dashboard_consent_notification(request, user, course_enrollments)
-
-    # Disable lookup of Enterprise consent_required_course due to ENT-727
-    # Will re-enable after fixing WL-1315
+    enterprise_customer = enterprise_customer_for_request(request)
     consent_required_courses = set()
     enterprise_customer_name = None
+    if enterprise_customer:
+        course_consents = consent_needed_for_courses(request.user,
+                                                     [str(enrollment.course_id) for enrollment in course_enrollments])
+        consent_required_courses = {
+             enrollment.course_id for enrollment in course_enrollments
+             if course_consents[str(enrollment.course_id)] is True
+         }
```

Comment:
actually wondering if it would be better to have the new api  just return the list of courses that have consent required so we don't need to have two loops.

something like get_consent_required_courses(user, course_ids) that returns a set of the course_ids that have consent required and ignores the others